nix/update.sh

```diff
@@ -1,53 +1,38 @@
-#!/usr/bin/env nix-shell
-#!nix-shell -i bash -p coreutils git jq nix-prefetch-github nix-prefetch-git
+#!/usr/bin/env bash
 
-set -euo pipefail
+set -eu
 
 dir=$(dirname "$(readlink -f "$0")")
-toplevel=$(git -C "${dir}" rev-parse --show-toplevel)
-mainroot="$(git -C "${dir}" rev-parse --path-format=absolute --show-toplevel)"
 
-update-source-github() {
-  local _name="$1"
-  local _submodule="$2"
-  local _user="$3"
-  local _repo="$4"
-  local _path="${toplevel}/${_submodule}"
-  local _rev _version
-  _rev="$(git -C "${toplevel}" submodule status -- "${_submodule}" | cut -c2- | cut -d' ' -f 1)"
-  git -C "${toplevel}" submodule update --init "${_submodule}"
-  _version="$(git -C "${_path}" describe --tag)"
+"${dir}"/update-plugins.sh
 
-  nix-prefetch-github --no-fetch-submodules --rev="${_rev}" "${_user}" "${_repo}" \
-    | jq --arg version "${_version}" '. + {$version}' \
-    > "${dir}/${_name}/source.json"
+command -v nix > /dev/null && {
+  echo "Updating with the local Nix installation..."
+  "${dir}/update-impl.sh"
+  exit $?
 }
 
-update-source() {
-  local _name="$1"
-  local _submodule="$2"
-  local _url
-  _url="$(git config --file "${mainroot}/.gitmodules" --get "submodule.${_submodule}.url")"
-  echo "updating ${_name} at ${_submodule} from ${_url}"
-
-  re="^((https?|ssh|git|ftps?):\/\/)?(([^\/@]+)@)?([^\/:]+)[\/:]([^\/:]+)\/(.+)\/?$"
-
-  if [[ "${_url}" =~ ${re} ]]; then
-    local _hostname=${BASH_REMATCH[5]}
-    local _user=${BASH_REMATCH[6]}
-    local _repo=${BASH_REMATCH[7]}
-
-    if [[ "${_hostname}" == "github.com" ]]; then
-      update-source-github "${_name}" "${_submodule}" "${_user}" "${_repo}"
-    else
-      >&2 echo "Updating submodules from ${_hostname} is not implemented"
-      exit 1
-    fi
+toplevel="$(git -C "${dir}" rev-parse --show-toplevel)"
+mainroot="$(git -C "${dir}" rev-parse --path-format=absolute --show-toplevel)"
+gitdir="$(git -C "${dir}" rev-parse --path-format=absolute --git-common-dir)"
+echo "toplevel = ${toplevel}"
+echo "mainroot = ${mainroot}"
+echo "gitdir = ${gitdir}"
+
+command -v docker > /dev/null && {
+  echo "Updating with with a nix docker container..."
```

Comment:
When I run this, it replaces the plugins/version.json file with this for me:

```diff
M nix/vast/plugins/versions.nix
@@ -1,6 +1,6 @@
 {
-  compaction = "1.4.0";
-  inventory = "1.0.0";
-  matcher = "3.0.0";
-  netflow = "1.0.0";
+  compaction = "";
+  inventory = "";
+  matcher = "";
+  netflow = "";
 }
```

Reply:
Was the submodule initialized when you tried that?

Reply:
Yes, although it was not a fresh checkout. Let me try again after recreating the worktree, because I had to meddle with the submodule in my first review and I wanna make sure that I didn't mess this up because of that.

Reply:
I traced it down with `set -x`, this is always returning an empty string for me:

```
version="$(sed -n 's|[^(]VERSION\s\+\([0-9A-Za-z\.-_]\+\)$|\1|p' "$cm" | tr -d '[:space:]')"
```

I'll get you a patch in a minute that fixes this.

Reply:
Actually, considering this file should not exist as of the follow-up PR, maybe we should just leave that out. Ideally, that non-portable sed-call should not be needed.

Reply:
```diff
diff --git a/nix/update-plugins.sh b/nix/update-plugins.sh
index e36fbee19..0860f62b3 100755
--- a/nix/update-plugins.sh
+++ b/nix/update-plugins.sh
@@ -24,7 +24,12 @@ echo "Extracting plugin versions..."
     # Extract the plugin version from CMakeLists.txt.
     # TODO: Remove this and only store the plugin names when we no longer print
     # plugin versions in `vast version`.
-    version="$(sed -n 's|[^(]VERSION\s\+\([0-9A-Za-z\.-_]\+\)$|\1|p' "$cm" | tr -d '[:space:]')"
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+      SED=gsed
+    else
+      SED=sed
+    fi
+    version="$("${SED}" -n 's|[^(]VERSION\s\+\([0-9A-Za-z\.-_]\+\)$|\1|p' "$cm" | tr -d '[:space:]')"
     echo "plugin = $plugin, version = $version" >&2
     echo "  $plugin = \"$version\";"
   done
```

Here's a hack-ish workaround